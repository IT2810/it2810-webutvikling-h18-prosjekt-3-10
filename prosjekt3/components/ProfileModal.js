import React from 'react';
import {
    StyleSheet,
    View,
    Modal,
    TouchableHighlight,
    KeyboardAvoidingView,
} from 'react-native';
import { Icon, } from 'react-native-elements';
import ProfileScreen from '../screens/ProfilesScreen';

export default class ProfileModal extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            modalVisible: false,
        }
    }

    setModalVisible(visible) {
        this.setState({ modalVisible: visible });
    }

    ProfileMod = () => {
        return (
            <View>

                {/* the modal for profile*/}
                <Modal
                    animationType="slide"
                    transparent={false}
                    visible={this.state.modalVisible}
                    presentationStyle="formSheet"
                    onRequestClose={() => { // required on android, making it posible to use hardware back-button
                        this.setModalVisible(!this.state.modalVisible);
                    }}>
                    <View >
                        {/* get return screen */}
                        <ProfileScreen />

                        <TouchableHighlight style={styles.closeModalButton}
                            underlayColor={'transparent'}
                            onPress={() => { // closing modal when close-button is pressed
                                this.setModalVisible(!this.state.modalVisible);
                            }}>
                            {/* home button in profile screen */}
                            <Icon
                                reverse
                                type='font-awesome'
                                name='home'
                                color="#cc0000"
                                size={30}
                            />
                        </TouchableHighlight>
                    </View>
                </Modal>

                <TouchableHighlight
                    underlayColor={'transparent'}
                    onPress={() => { // opens modal when profile-object is pressed
                        this.setModalVisible(true);
                    }}>
                    <View>
                        {/* profile button on home screen */}
                        <Icon style={styles.modalProfile}
                            reverse
                            type='font-awesome'
                            name='user'
                            color='#5F7C80'
                            size={40}
                        />
                    </View>
                </TouchableHighlight>
            </View>
        )
    }
    render() {
        return (
            <View>
                {this.ProfileMod()}
            </View>
        )
    }
}

const styles = StyleSheet.create({

    closeModalButton: {
        position: 'absolute',
        top: "50%",
        right: 2,
    },

    modalProfile: {
        position: 'absolute',
        bottom: 3,
    },
})