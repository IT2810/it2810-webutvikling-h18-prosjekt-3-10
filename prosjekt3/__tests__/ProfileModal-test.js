import React from 'react';
import renderer from 'react-test-renderer';
import ProfileModal from '../components/ProfileModal';
import NavigationTestUtils from 'react-navigation/NavigationTestUtils';

// snapshot test
const release = () => jest.unmock('AsyncStorage')
describe('Profile-modal snapshot', () => {
    jest.useFakeTimers();
    beforeEach(() => {
        NavigationTestUtils.resetInternalState();
    });

    it('renders the Profile-modal', async () => {
        const tree = renderer.create(<ProfileModal />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});

release();