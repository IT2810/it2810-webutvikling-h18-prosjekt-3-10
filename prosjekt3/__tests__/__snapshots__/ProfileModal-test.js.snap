// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile-modal snapshot renders the Profile-modal 1`] = `
<View>
  <View>
    <Modal
      animationType="slide"
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      presentationStyle="formSheet"
      transparent={false}
      visible={false}
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#476DC5",
              "borderBottomColor": "#f2f2f2",
              "borderBottomWidth": 1,
              "height": 70,
              "padding": 15,
            },
            Object {
              "backgroundColor": "#fff",
            },
            Object {
              "borderBottomColor": "black",
              "borderBottomWidth": 2,
              "height": 100,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "flex-end",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "space-between",
              },
              Object {
                "marginTop": 13,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "height": 24,
                "width": 24,
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {},
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Object {
                  "color": "black",
                  "fontSize": 20,
                  "fontVariant": Array [
                    "small-caps",
                  ],
                  "fontWeight": "600",
                },
              ]
            }
          >
            profile
          </Text>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                false,
                false,
                false,
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#2f95dc",
                    "fontSize": 28,
                  },
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                    undefined,
                  ],
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
      <View>
        <View
          enabled={true}
          keyboardVerticalOffset={0}
          onLayout={[Function]}
        >
          <View
            style={
              Array [
                undefined,
                Object {
                  "bottom": 0,
                },
              ]
            }
          />
        </View>
      </View>
    </Modal>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={null}
    >
      <View
        style={null}
      >
        <View
          style={
            Array [
              Object {
                "margin": 7,
              },
              Object {
                "borderRadius": 44,
                "height": 84,
                "width": 84,
              },
              false,
              Object {
                "alignItems": "center",
                "backgroundColor": "#425b84",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
          underlayColor="#425b84"
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "white",
                  "fontSize": 40,
                },
                Array [
                  Object {
                    "backgroundColor": "transparent",
                  },
                  undefined,
                ],
                Object {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
